aten/src/ATen/native/native_functions.yaml

```diff
@@ -3759,14 +3759,14 @@
 - func: narrow_copy(Tensor self, int dim, SymInt start, SymInt length) -> Tensor
   variants: function, method
   dispatch:
-    CPU: narrow_copy_dense_cpu
     SparseCPU, SparseCUDA: narrow_copy_sparse
     CompositeExplicitAutogradNonFunctional: narrow_copy_dense
   tags: view_copy
 
 - func: narrow_copy.out(Tensor self, int dim, SymInt start, SymInt length, *, Tensor(a!) out) -> Tensor(a!)
+  variants: function
   dispatch:
-    CPU: narrow_copy_dense_cpu_out
+    CompositeExplicitAutogradNonFunctional: narrow_copy_out
```

Comment:
Heads up: changes dispatch for `out`. Seems to work (based on limited local testing), but not sure if okay.

Also, not for this PR: these variants are declared with `SymInt` while the kernels use `int`s. Probably will be addressed later, e.g., in [this PR](https://github.com/pytorch/pytorch/pull/84173), there's a comment:
> TODO: properly support SymInt narrow_copy

Reply:
What's the reason for this change? I don't think we should do this. If anything needs to be fixed, it should be fixed at the root, that is, in the autogen

Reply:
nvm, I misread the change. Yup, this LGTM. We don't need to implement this manually, the trivial implementation should do the job.

Reply:
Actually, your point of:
>Also, not for this PR: these variants are declared with SymInt while the kernels use ints. Probably will be addressed later, e.g., in https://github.com/pytorch/pytorch/pull/84173, there's a comment:

looks like it may be the root of this issue you found: https://github.com/pytorch/pytorch/pull/87045#discussion_r1008952196